python/tools/test_code_execution_security.py

```
@@ -0,0 +1,41 @@
+import unittest
+import os
+import sys
+import json
+from python.tools.code_execution_tool import execute_code
+
+class TestCodeExecutionSecurity(unittest.TestCase):
+    def test_environment_allowlist(self):
+        # Set a sensitive variable that should NOT be passed
+        os.environ["SECRET_TOKEN_123"] = "should-not-be-leaked"
```

Ruff (0.15.9)

[error] Possible hardcoded password assigned to: "SECRET_TOKEN_123"

(S105)